remote/remote.go

```diff
@@ -24,17 +27,19 @@ import (
 type Image struct {
 	keychain   authn.Keychain
 	repoName   string
+	platform   imgutil.Platform
```

Comment:
It feels a little weird to me that we have this field in the Image struct that could be inaccurate. That would happen if
1. A user called `SetOS`
2. No platform was provided but the base image is a windows image

Someone could easily introduce a bug if they assume this is always set correctly.

Reply:
I removed `platform` from the `Image`, as suggested, with the latest set of commits.